systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1077,453 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+            .endKafkaClientAuthenticationTls()
+            .withNewTls()
+                .withTrustedCertificates(certSecretSource)
+            .endTls()
+            .build();
+
+        KafkaMirrorMaker2ClusterSpec targetClusterWithTlsAuth = new KafkaMirrorMaker2ClusterSpecBuilder()
+            .withAlias(kafkaClusterTargetName)
+            .withBootstrapServers(KafkaResources.tlsBootstrapAddress(kafkaClusterTargetName))
+            .withNewKafkaClientAuthenticationTls()
+                .withNewCertificateAndKey()
+                    .withSecretName(kafkaUserTargetName)
+                    .withCertificate("user.crt")
+                    .withKey("user.key")
+                .endCertificateAndKey()
+            .endKafkaClientAuthenticationTls()
+            .withNewTls()
+                .withTrustedCertificates(certSecretTarget)
+            .endTls()
+            .addToConfig("config.storage.replication.factor", -1)
+            .addToConfig("offset.storage.replication.factor", -1)
+            .addToConfig("status.storage.replication.factor", -1)
+            .build();
+
+        resourceManager.createResource(extensionContext, KafkaMirrorMaker2Templates.kafkaMirrorMaker2(clusterName, kafkaClusterTargetName, kafkaClusterSourceName, 1, true)
+            .editSpec()
+                .withClusters(sourceClusterWithTlsAuth, targetClusterWithTlsAuth)
+                .editFirstMirror()
+                    .withTopicsPattern(MIRRORMAKER2_TOPIC_NAME + ".*")
+                        .editSourceConnector()
+                            .addToConfig("refresh.topics.interval.seconds", 1)
+                        .endSourceConnector()
+                .endMirror()
+            .endSpec()
+            .build());
+        Map<String, String> mmSnapshot = DeploymentUtils.depSnapshot(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName));
+
+        internalKafkaClient = internalKafkaClient.toBuilder()
+            .withTopicName(topicSourceNameA)
+            .withClusterName(kafkaClusterSourceName)
+            .withKafkaUsername(userSource.getMetadata().getName())
+            .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+            .build();
+
+        int sent = internalKafkaClient.sendMessagesTls();
+        internalKafkaClient.checkProducedAndConsumedMessages(sent, internalKafkaClient.receiveMessagesTls());
+
+        internalKafkaClient = internalKafkaClient.toBuilder()
+            .withTopicName(topicTargetNameA)
+            .withClusterName(kafkaClusterTargetName)
+            .withKafkaUsername(userTarget.getMetadata().getName())
+            .build();
+
+        LOGGER.info("Consumer in target cluster and topic should receive {} messages", messagesCount);
+        internalKafkaClient.checkProducedAndConsumedMessages(sent, internalKafkaClient.receiveMessagesTls());
+        LOGGER.info("Messages successfully mirrored");
+
+        LOGGER.info("Going to delete ClientsCA certificates");
```

Comment:
Wouldn't it be better to use annotation to renew them or delete just the user secret? Deleting the CA secret does not seem like the safe way to do this.

Reply:
I do both ways. Removal of certs and forcing annotation. Do you think it's unnecessary too much?

Reply:
Well, but 40 lines below it is for ClusterCA, or? I would use the annotation for both I guess => that is the clean approach and you do not test here what happens when the CA secret is deleted but to test MM2 rolls.